homeassistant/components/bayesian/config_flow.py

```diff
@@ -0,0 +1,620 @@
+    # In this case, we want to remove nested items so we update the options directly.
+    # Remove the last indexes first so subsequent items to be removed aren't shifted
+    for index in sorted(indexes, reverse=True):
+        observations.pop(index)
+
+    _LOGGER.debug("Deleted observations: '%s'", observations)
+    return {}
+
+
+CONFIG_FLOW: dict[str, SchemaFlowMenuStep | SchemaFlowFormStep] = {
+    str(ConfigFlowSteps.USER): SchemaFlowFormStep(
+        CONFIG_SCHEMA,
+        validate_user_input=_validate_user,
+        next_step=ConfigFlowSteps.OBSERVATION_SELECTOR,
+    ),
+    str(ConfigFlowSteps.OBSERVATION_SELECTOR): SchemaFlowMenuStep(
+        ObservationTypes.list()
+    ),
+    str(ObservationTypes.STATE): SchemaFlowFormStep(
+        _get_state_schema,
+        next_step=_add_more_or_end,
+        validate_user_input=_validate_observation_setup,
+        suggested_values=_get_observation_values_if_editing,
+    ),
+    str(ObservationTypes.NUMERIC_STATE): SchemaFlowFormStep(
+        _get_numeric_state_schema,
+        next_step=_add_more_or_end,
+        validate_user_input=_validate_observation_setup,
+        suggested_values=_get_observation_values_if_editing,
+    ),
+    str(ObservationTypes.TEMPLATE): SchemaFlowFormStep(
+        _get_template_schema,
+        next_step=_add_more_or_end,
+        preview=str(ObservationTypes.TEMPLATE),
+        validate_user_input=_validate_observation_setup,
+        suggested_values=_get_observation_values_if_editing,
+    ),
+}
+
+OPTIONS_FLOW: dict[str, SchemaFlowMenuStep | SchemaFlowFormStep] = {
+    str(OptionsFlowSteps.INIT): SchemaFlowMenuStep(
+        OptionsFlowSteps.list_primary_steps()
+    ),
+    str(OptionsFlowSteps.BASE_OPTIONS): SchemaFlowFormStep(
+        OPTIONS_SCHEMA,
+        suggested_values=_get_base_suggested_values,
+        validate_user_input=_validate_user,
+    ),
+    str(OptionsFlowSteps.SELECT_EDIT_OBSERVATION): SchemaFlowFormStep(
+        _get_select_observation_schema,
+        suggested_values=None,
+        next_step=_get_flow_step_for_editing,
+    ),
+    str(OptionsFlowSteps.REMOVE_OBSERVATION): SchemaFlowFormStep(
+        _get_remove_observation_schema,
+        suggested_values=None,
+        validate_user_input=_validate_remove_observation,
+    ),
+}
+OPTIONS_FLOW.update(CONFIG_FLOW)
+
+
+class BayesianConfigFlowHandler(SchemaConfigFlowHandler, domain=DOMAIN):
+    """Bayesian config flow."""
+
+    # The schema version of the entries that it creates
+    # Home Assistant will call your migrate method if the version changes
+    VERSION = 1
+    MINOR_VERSION = 1
+
+    config_flow = CONFIG_FLOW
+    options_flow = OPTIONS_FLOW
+
+    def async_config_entry_title(self, options: Mapping[str, str]) -> str:
+        """Return config entry title."""
+        name: str = options[CONF_NAME]
+        return name
+
+    @staticmethod
+    async def async_setup_preview(hass: HomeAssistant) -> None:
```

Comment:
I think we should remove preview from this PR and do it in a follow-up as it's quite big enough as it is.

Reply:
done now